sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -855,6 +855,12 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
     }
   }
 
+  test("SPARK-36726: test incorrect Parquet row group file offset") {
+    readParquetFile("test-data/malformed-file-offset.parquet") { df =>
+      assert(df.count() == 3650)
```

Comment:
should we test for both vec/non vec code path?

Reply:
I think `readParquetFile` already takes care of that, see `FileBasedDataSourceTest.readFile`.

Reply:
I thought vectorized is enabled by default, and non-vectorized path will not be tested.  `  protected def readFile(path: String, testVectorized: Boolean = true)`, so we need

```
  Seq(true, false).foreach { vec =>
      readParquetFile(testFile("file.parquet"), vec) { df =>
         ...
     }
  }
```

Reply:
This is how `FileBasedDataSourceTest.readFile` is defined:

```scala
  protected def readFile(path: String, testVectorized: Boolean = true)
      (f: DataFrame => Unit): Unit = {
    withSQLConf(vectorizedReaderEnabledKey -> "false") {
      f(spark.read.format(dataSourceName).load(path.toString))
    }
    if (testVectorized) {
      withSQLConf(vectorizedReaderEnabledKey -> "true") {
        f(spark.read.format(dataSourceName).load(path.toString))
      }
    }
  }
```
so to me it tests both paths. Did I miss anything?

Reply:
I am surprised that the non-vectorized reader is always tested with/without testVectorized enabled...
```
  protected def readFile(path: String, testVectorized: Boolean = true)
      (f: DataFrame => Unit): Unit = {
    withSQLConf(vectorizedReaderEnabledKey -> "false") {
      f(spark.read.format(dataSourceName).load(path.toString))
    }
    if (testVectorized) {
      withSQLConf(vectorizedReaderEnabledKey -> "true") {
        f(spark.read.format(dataSourceName).load(path.toString))
      }
    }
  }
```

Reply:
Yes, I guess it makes sense since in most cases we'd want to test both code paths and this is convenient for that.

Reply:
You are right. I'm surprised that non-vectorized path is always tested. In many tests, we specifically test both paths, and they are not necessary given the non-vectorized will always be tested.